Manager/SubscriberManager.php

```diff
@@ -0,0 +1,194 @@
+use mbzSubscriber\Unsubscribe;
+use MindbazBundle\Model\Subscriber;
+use MindbazBundle\Serializer\SubscriberEncoder;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * @author Vincent Chalamon <vincent@les-tilleuls.coop>
+ */
+class SubscriberManager
+{
+    const MINDBAZ_SEND_RESPONSE_OK = 'OK';
+    const MINDBAZ_SEND_RESPONSE_NOK = 'NOK';
+
+    /**
+     * @var SubscriberWebService
+     */
+    private $subscriberWebService;
+
+    /**
+     * @var OneshotWebService
+     */
+    private $oneshotWebService;
+
+    /**
+     * @var SerializerInterface|DenormalizerInterface
+     */
+    private $serializer;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param SubscriberWebService                       $subscriberWebService
+     * @param OneshotWebService                          $oneshotWebService
+     * @param SerializerInterface|DenormalizerInterface  $serializer
+     * @param LoggerInterface|null                       $logger
+     */
+    public function __construct(SubscriberWebService $subscriberWebService, OneshotWebService $oneshotWebService, SerializerInterface $serializer, LoggerInterface $logger = null)
+    {
+        $this->subscriberWebService = $subscriberWebService;
+        $this->oneshotWebService = $oneshotWebService;
+        $this->serializer = $serializer;
+        $this->logger = $logger ?: new NullLogger();
+    }
+
+    /**
+     * @param array $fields
+     *
+     * @return Subscriber
+     */
+    public function create(array $fields)
+    {
+        $subscriber = $this->serializer->denormalize($fields, Subscriber::class);
+        $this->insert($subscriber);
+
+        return $subscriber;
+    }
+
+    /**
+     * @param Subscriber $subscriber
+     */
+    public function insert(Subscriber $subscriber)
+    {
+        /** @var MindbazSubscriber $mbzSubscriber */
+        $mbzSubscriber = $this->serializer->serialize($subscriber, SubscriberEncoder::FORMAT);
+        $result = $this->subscriberWebService->InsertSubscriber(new InsertSubscriber($mbzSubscriber, true));
+        $subscriber->setId($result->getInsertSubscriberResult());
+
+        $this->logger->info('New subscriber inserted in Mindbaz', ['id' => $subscriber->getId()]);
+    }
+
+    /**
+     * @param Subscriber $subscriber
+     */
+    public function unsubscribe(Subscriber $subscriber)
+    {
+        $result = $this->subscriberWebService->Unsubscribe(new Unsubscribe($subscriber->getId(), null, null));
+        if (true === $result->getUnsubscribeResult()) {
+            $this->logger->info('Subscriber successfully unsubscribed', ['id' => $subscriber->getId()]);
+        } else {
+            $this->logger->error('An error occurred while unsubscribing subscriber', ['id' => $subscriber->getId(), 'response' => $result->getUnsubscribeResult()]);
```

Comment:
move the previous comment for send method
Throw exception for unsubscribe method is useless